Lib/test/test_smtplib.py

```diff
@@ -920,57 +924,78 @@ def testEXPN(self):
-        smtp = smtplib.SMTP(HOST, self.port, local_hostname='localhost', timeout=15)
-        resp = smtp.login(sim_auth[0], sim_auth[1])
-        self.assertEqual(resp, (235, b'Authentication Succeeded'))
-        smtp.close()
+        for sim_auth in sim_auth_lists:
+            with self.subTest(sim_auth=sim_auth):
+                smtp = smtplib.SMTP(HOST, self.port, local_hostname='localhost', timeout=15)
+                SimSMTPChannel.sim_auth = sim_auth
+                resp = smtp.login(sim_auth[0], sim_auth[1])
+                self.assertEqual(resp, (235, b'Authentication Succeeded'))
+                smtp.close()
 
     def testAUTH_CRAM_MD5(self):
         self.serv.add_feature("AUTH CRAM-MD5")
-        smtp = smtplib.SMTP(HOST, self.port, local_hostname='localhost', timeout=15)
-        resp = smtp.login(sim_auth[0], sim_auth[1])
-        self.assertEqual(resp, (235, b'Authentication Succeeded'))
-        smtp.close()
+        for sim_auth in sim_auth_lists:
+            with self.subTest(sim_auth=sim_auth):
+                smtp = smtplib.SMTP(HOST, self.port, local_hostname='localhost', timeout=15)
+                SimSMTPChannel.sim_auth = sim_auth
+                resp = smtp.login(sim_auth[0], sim_auth[1])
+                self.assertEqual(resp, (235, b'Authentication Succeeded'))
+                smtp.close()
 
     def testAUTH_multiple(self):
         # Test that multiple authentication methods are tried.
         self.serv.add_feature("AUTH BOGUS PLAIN LOGIN CRAM-MD5")
-        smtp = smtplib.SMTP(HOST, self.port, local_hostname='localhost', timeout=15)
-        resp = smtp.login(sim_auth[0], sim_auth[1])
-        self.assertEqual(resp, (235, b'Authentication Succeeded'))
-        smtp.close()
+        for sim_auth in sim_auth_lists:
+            with self.subTest(sim_auth=sim_auth):
+                smtp = smtplib.SMTP(HOST, self.port, local_hostname='localhost', timeout=15)
+                SimSMTPChannel.sim_auth = sim_auth
+                resp = smtp.login(sim_auth[0], sim_auth[1])
+                self.assertEqual(resp, (235, b'Authentication Succeeded'))
+                smtp.close()
 
     def test_auth_function(self):
         supported = {'CRAM-MD5', 'PLAIN', 'LOGIN'}
         for mechanism in supported:
             self.serv.add_feature("AUTH {}".format(mechanism))
         for mechanism in supported:
             with self.subTest(mechanism=mechanism):
-                smtp = smtplib.SMTP(HOST, self.port,
-                                    local_hostname='localhost', timeout=15)
-                smtp.ehlo('foo')
-                smtp.user, smtp.password = sim_auth[0], sim_auth[1]
-                method = 'auth_' + mechanism.lower().replace('-', '_')
-                resp = smtp.auth(mechanism, getattr(smtp, method))
-                self.assertEqual(resp, (235, b'Authentication Succeeded'))
-                smtp.close()
+                for sim_auth in sim_auth_lists:
```

Comment:
The `with self.subTest(...)` line should be in the internal loop, i.e. `with self.subTest(mechanism=mechanism, sim_auth=sim_auth):`